### PR TITLE
Support `Upload` and `Upgrade` transactions

### PR DESCRIPTION
**Pull request by xgreenx**
The change just adda basic support for `Upload` and `Upgrade` transactions. It already allows re-usage of all estimation and sending logic which is used in the https://github.com/FuelLabs/fuel-utils.

### Checklist
- [ ] I have linked to any relevant issues.
- [ ] I have updated the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by segfault-magnet — COMMENTED** [review]
We need to add `UploadSubsection` and `UpgradePurpose` somewhere inside `fuels::...` reexports.

a `pub use` in `transaction_builders` is going to propagate it into the `fuels::prelude` and have it be available under `fuels::types::transaction_builders::...`

Also we should add e2e tests to verify that this works, if that's possible @xgreenx ?

**Comment by xgreenx** [comment]
@segfault-magnet May I ask you to make this PR fulfill your standards of the codebase, please?=)

The example of how I used `Upload` and `Upgrade` transactions you can find in [this](https://github.com/FuelLabs/fuel-utils/blob/master/network-upgrader/src/main.rs#L136) repository.

**Comment by segfault-magnet** [comment]
> @segfault-magnet May I ask you to make this PR fulfill your standards of the codebase, please?=)
> 
> The example of how I used `Upload` and `Upgrade` transactions you can find in [this](https://github.com/FuelLabs/fuel-utils/blob/master/network-upgrader/src/main.rs#L136) repository.
> 

Sure. Tnx for the PR and the examples, I'll finish it up.

**Review by MujkicA — APPROVED** [review]

File: packages/fuels-core/src/types/transaction_builders.rs [hunk 1]
```diff
@@ -716,6 +809,189 @@ impl CreateTransactionBuilder {
...
+            .with_tx_policies(tx_policies)
+            .with_purpose(UpgradePurpose::StateTransition { root })
+    }
+
+    pub fn prepare_consensus_parameters_upgrade(
+        consensus_parameters: &ConsensusParameters,
+        tx_policies: TxPolicies,
+    ) -> Self {
+        let serialized_consensus_parameters = postcard::to_allocvec(consensus_parameters)
+            .expect("Impossible to fail unless it not enough memory");
```
Comment (digorithm) [inline packages/fuels-core/src/types/transaction_builders.rs:965]:
```suggestion
            .expect("Impossible to fail unless there is not enough memory");
```

**Review by digorithm — CHANGES_REQUESTED** [review]
Just the tiniest typo. Other than that, it looks great!

**Review by hal3e — APPROVED** [review]

**Review by Salka1988 — APPROVED** [review]

**Pull request merged**